react-example/src/components/LoginForm.tsx

```diff
@@ -42,12 +55,14 @@ export const LoginForm: FC<Props> = ({ setEmail, logout, refreshJwt }) => {
   const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
     e.preventDefault();
 
-    setEmail(email)
+    const setUser = useEmail ? setEmail : setUserId;
+
+    setUser(user)
       .then(() => {
         setEditingUser(false);
-        setLoggedInUser({ type: 'user_update', data: email });
+        setLoggedInUser({ type: 'user_update', data: user });
       })
-      .catch(() => updateEmail('Something went wrong!'));
+      .catch(() => updateUser('Something went wrong!'));
```

Comment:
Looking at this, it's kinda strange that we update the user value to this error message. Can consider refactoring at some point to show an inline error below the field or open an alert or something. Just a thought

Reply:
yeah, great idea